projects/angular-sticky-things/src/lib/sticky-thing.directive.ts

```diff
@@ -188,56 +182,81 @@ export class StickyThingDirective implements OnInit, AfterViewInit, OnDestroy {
     return el.getBoundingClientRect();
   }
 
-  private determineStatus(originalVals: StickyPositions, pageYOffset: number, marginTop: number, marginBottom: number, enabled: boolean): StickyStatus {
-    const stickyElementHeight = this.getComputedStyle(this.stickyElement.nativeElement).height;
-    const reachedLowerEdge = this.boundaryElement && window.pageYOffset + stickyElementHeight + marginBottom >= (originalVals.bottomBoundary - marginTop);
+  private determineStatus(enabled: boolean, pageYOffset: number) {
+    const elementPos = this.getOriginalPosition();
+    const isSticky = enabled && pageYOffset + this.marginTop$.value > elementPos.offsetY;
     return {
-      isSticky: enabled && pageYOffset > originalVals.offsetY,
-      reachedLowerEdge,
-      marginBottom,
-      marginTop,
+      isSticky,
     };
+
   }
 
 
+  // not always pixel. e.g. ie9
+  private getMargins(): { top: number, bottom: number } {
+    const stickyStyles = window.getComputedStyle(this.stickyElement.nativeElement);
+    const top = parseInt(stickyStyles.marginTop, 10);
+    const bottom = parseInt(stickyStyles.marginBottom, 10);
+    return {top, bottom};
+  }
+
   /**
    * Gets the offset for element. If the element
    * currently is sticky, it will get removed
    * to access the original position. Other
    * wise this would just be 0 for fixed elements. */
-  private determineElementOffsets(): StickyPositions {
-    if (this.sticky) {
-      this.removeSticky();
-    }
+  private getOriginalPosition(): StickyPositions {
 
-    let bottomBoundary: number | null = null;
+    const margins = this.getMargins();
 
-    if (this.boundaryElement) {
-      const boundaryElementHeight = this.getComputedStyle(this.boundaryElement).height;
-      const boundaryElementOffset = getPosition(this.boundaryElement).y;
-      bottomBoundary = boundaryElementHeight + boundaryElementOffset;
+    if (this.sticky) {
+      this.removeSticky();
     }
 
-    return {offsetY: (getPosition(this.stickyElement.nativeElement).y - this.marginTop$.value), bottomBoundary};
+    return {
+      offsetY: (getPosition(this.stickyElement.nativeElement).y - margins.top),
+    };
   }
 
-  private makeSticky(boundaryReached: boolean = false, marginTop: number, marginBottom: number): void {
+  private prepareSticky(): void {
+
+    const marginTop = this.marginTop$.value;
+    const marginBottom = this.marginBottom$.value;
+    const stickyElStyle = this.getComputedStyle(this.stickyElement.nativeElement);
+    const boundaryElStyle = this.getComputedStyle(this.boundaryElement);
```

Comment:
This will break if you have no `boundaryElement`.

Reply:
Ah, good catch, more tests and strict mode would help..